test/libsolidity/semanticTests/externalContracts/FixedFeeRegistrar.sol

```diff
@@ -0,0 +1,98 @@
+}
+
+contract FixedFeeRegistrar is Registrar {
+	struct Record {
+		address addr;
+		address subRegistrar;
+		bytes32 content;
+		address owner;
+	}
+
+	modifier onlyrecordowner(string memory _name) { if (m_record(_name).owner == msg.sender) _; }
+
+	function reserve(string memory _name) public payable {
+		Record storage rec = m_record(_name);
+		if (rec.owner == 0x0000000000000000000000000000000000000000 && msg.value >= c_fee) {
+			rec.owner = msg.sender;
+			emit Changed(_name);
+		}
+	}
+	function disown(string memory _name, address payable _refund) onlyrecordowner(_name) public {
+		delete m_recordData[uint(keccak256(bytes(_name))) / 8];
+		if (!_refund.send(c_fee))
+			revert();
+		emit Changed(_name);
+	}
+	function transfer(string memory _name, address _newOwner) onlyrecordowner(_name) public {
+		m_record(_name).owner = _newOwner;
+		emit Changed(_name);
+	}
+	function setAddr(string memory _name, address _a) onlyrecordowner(_name) public {
+		m_record(_name).addr = _a;
+		emit Changed(_name);
+	}
+	function setSubRegistrar(string memory _name, address _registrar) onlyrecordowner(_name) public {
+		m_record(_name).subRegistrar = _registrar;
+		emit Changed(_name);
+	}
+	function setContent(string memory _name, bytes32 _content) onlyrecordowner(_name) public {
+		m_record(_name).content = _content;
+		emit Changed(_name);
+	}
+
+	function record(string memory _name) public view returns (address o_addr, address o_subRegistrar, bytes32 o_content, address o_owner) {
+		Record storage rec = m_record(_name);
+		o_addr = rec.addr;
+		o_subRegistrar = rec.subRegistrar;
+		o_content = rec.content;
+		o_owner = rec.owner;
+	}
+	function addr(string memory _name) public override view returns (address) { return m_record(_name).addr; }
+	function subRegistrar(string memory _name) public override view returns (address) { return m_record(_name).subRegistrar; }
+	function content(string memory _name) public override view returns (bytes32) { return m_record(_name).content; }
+	function owner(string memory _name) public override view returns (address) { return m_record(_name).owner; }
+
+	Record[2**253] m_recordData;
+	function m_record(string memory _name) view internal returns (Record storage o_record) {
+		return m_recordData[uint(keccak256(bytes(_name))) / 8];
+	}
+	uint constant c_fee = 69 ether;
+}
+// ====
+// compileViaYul: also
+// ----
+// constructor()
+// gas irOptimized: 544916
+// gas legacy: 942808
+// gas legacyOptimized: 492291
+// reserve(string), 69 ether: 0x20, 3, "abc" ->
+// gas irOptimized: 45527
+// gas legacy: 46242
+// gas legacyOptimized: 45491
+// owner(string): 0x20, 3, "abc" -> 0x1212121212121212121212121212120000000012
+// reserve(string), 70 ether: 0x20, 3, "def" ->
+// owner(string): 0x20, 3, "def" -> 0x1212121212121212121212121212120000000012
+// reserve(string), 68 ether: 0x20, 3, "ghi" ->
+// owner(string): 0x20, 3, "ghi" -> 0
+// account: 1 -> 0x1212121212121212121212121212120000001012
```

Comment:
This is not 100% complete yet, such as the properties and disown test is missing, however without having comments/separation it is quite hard to follow, so would need #11427.

Reply:
Maybe you could at least use the `#` comments to mark where tests from the original start/end?

As far as I can see, you translated `creation`, `reserve`, `transfer` and `disown` (the latter two with slight modifications) and dropped `double_reserve` and `properties`?

Reply:
Only properties is missing. Everything else is there. I tried adding the `#` comments but they only work in specific parts and wherever I put that I got a parsing error.

Reply:
`double_reserve` seems missing too.

Well, there's this part but if that's it then it's just half of it:
```solidity
// reserve(string), 69 ether: 0x20, 3, "abc" ->
// owner(string): 0x20, 3, "abc" -> 0x1212121212121212121212121212120000000012
// account: 0 -> 0x1212121212121212121212121212120000000012
```
But I see that the other half requires sending ether form another account. I guess that's not currently possible in these tests?

Reply:
`double_reserve` is not missing, just that the C++ version *reset* the state every single test case and we don't.